### PR TITLE
06 09 2023  as delete app registration

### PR DESCRIPTION
**Pull request by AcceleryntSecurityDev**
Change(s):
   - New playbook

   Reason for Change(s):
   - New playbook

   Testing Completed:
   - Yes

image

### CODE REVIEW

**Review by rahul0216 — COMMENTED** [review]
Playbook looks good. Couple of suggestions,
1. I think In the readme file, this need to be highlighted as warning that all the app registration with the same name may get deleted.
2. Can you also add comment to incident via logic app, so that SOC engineer get to know that action is completed and these app registration has been deleted?

**Comment by AcceleryntSecurityDev** [comment]
@rahul0216 I have implemented your suggestions, noting that adding comments to incidents requires the playbook to have the microsoft sentinel contributor role, so I have added documentation for this as well.

File: Playbooks/AS-Delete-App-Registration/README.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+[![Deploy to Azure Gov](https://aka.ms/deploytoazuregovbutton)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FPlaybooks%2FAS-Delete-App-Registration%2Fazuredeploy.json)       
+
+This playbook is intended to be run from a Microsoft Sentinel incident. If any app registration entities are found (i.e. any entities where kind == CloudApplication), they will be deleted. This playbook matches by name, since a unique app registration ID cannot currently be pulled into the entity list, so if there are multiple app registrations exactly matching the name(s) of the CloudApplication entities, all will be deleted. The deleted app registration(s) will be noted in incident comment(s).
+
+> **Warning**
+> If there are multiple app registrations exactly matching the name(s) of the CloudApplication entities in the Microsoft icident used to run this playbook, all of those app registrations will be deleted
```
Comment (rahul0216) [inline Playbooks/AS-Delete-App-Registration/README.md:13]: Correct the spelling mistake 'icident'

**Review by rahul0216 — COMMENTED** [review]
Changes look good. There is one spelling mistake in readme files. Please correct that.

**Review by rahul0216 — COMMENTED** [review]
Changes look good to merge.

**Review by rahul0216 — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**